server/channels/api4/channel.go

```diff
@@ -1406,7 +1406,9 @@ func getChannelByName(c *Context, w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	} else {
-		if !c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), channel.Id, model.PermissionReadChannel) {
+		// allows team admins to access private channel
+		if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), channel.TeamId, model.PermissionManageTeam) &&
+			!c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), channel.Id, model.PermissionReadChannel) {
```

Comment:
Just making a remark out loud: I don't understand why team admins are not getting permission to read all the channels from their team by default 🤔

Reply:
So I actually think the bug may be here.
https://github.com/mattermost/mattermost/blob/3b69d36643abd3c5074c5813791aed213c172faa/server/channels/app/authorization.go#L106
Instead of passing down the "permission" variable, the `model.PermissionManageTeam` should be passed down. However, that change would affect a large number of APIs and require a lot more time and testing and understanding. I have already spend quite a bit of mana. And its been that way for years. So I made the decision (right or wrong) to isolate the change for only this PR.